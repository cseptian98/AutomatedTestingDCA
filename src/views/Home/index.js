import React from 'react';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import TodoLists from 'components/TodoLists';
import TodoItems from 'components/TodoItems';
import User from 'components/User';

const Tab = createBottomTabNavigator();

const Home = () => {
  return (
    <Tab.Navigator
      initialRouteName="TodoList"
      tabBarOptions={{
        activeTintColor: '#424242',
        style: {
          backgroundColor: '#FFD500',
        },
      }}
      backBehavior="history">
      <Tab.Screen
        name="TodoList"
        component={TodoLists}
        options={{
          tabBarLabel: 'List',
          tabBarIcon: ({color, size}) => (
            <Icon name="playlist-edit" color={color} size={size} />
          ),
        }}
      />
      <Tab.Screen
        name="TodoItems"
        component={TodoItems}
        options={{
          tabBarLabel: 'Item',
          tabBarIcon: ({color, size}) => (
            <Icon name="note-multiple" color={color} size={size} />
          ),
        }}
      />
      <Tab.Screen
        name="User"
        component={User}
        options={{
          tabBarLabel: 'User',
          tabBarIcon: ({color, size}) => (
            <Icon name="account-circle" color={color} size={size} />
          ),
        }}
      />
    </Tab.Navigator>
  );
};

export default Home;
