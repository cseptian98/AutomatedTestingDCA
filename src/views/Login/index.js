import React, {useState, useEffect} from 'react'
import {
  Text,
  View,
  Image,
  TextInput,
  TouchableOpacity,
  StatusBar,
  ActivityIndicator,
  Alert,
} from 'react-native'
import styles from './Login.styles'
import {storeData, getData} from 'api/Local'
import axiosConfig, {setToken} from 'api/BaseConfig'
import {StackActions} from '@react-navigation/native'

const Login = ({navigation}) => {
  const [email, setEmail] = useState('')
  const [value, setValue] = useState(null)
  const [password, setPassword] = useState('')
  const [isLoading, setIsLoading] = useState(false)

  const readFromStorage = async () => {
    const data = await getData()
    setValue(data)
  }

  useEffect(() => {
    readFromStorage()
  }, [])

  if(value) {
    setToken(value.token)
    navigation.replace('Home')
  }

  const doLogin = () => {
    setIsLoading(true)
    const onSuccess = ({data}) => {
      setIsLoading(false)
      storeData(data)
      setToken(data.token)
      navigation.replace('Home')
    }

    const onFailure = data => {
      console.log('debug error', data)
      Alert.alert('Login Error', 'Error Message', [
        {
          text: 'Close',
          onPress: () => setIsLoading(false),
        },
      ])
    }

    axiosConfig
      .post('api/Auth/login', {email, password})
      .then(onSuccess)
      .catch(onFailure)
  }

  const toRegister = () => {
    const movePage = StackActions.push('Register')
    navigation.dispatch(movePage)
  }

  return (
    <View style={styles.container}>
      <Image style={styles.image} source={require('assets/images/login.png')} />

      <StatusBar style="auto" />
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Email"
          placeholderTextColor="#000"
          keyboardType="email-address"
          onChangeText={email => setEmail(email)}
        />
      </View>

      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Password"
          placeholderTextColor="#000"
          secureTextEntry
          onChangeText={password => setPassword(password)}
        />
      </View>
      {isLoading ? (
        <ActivityIndicator size="large" color="#0D47A1" />
      ) : (
        <>
          <TouchableOpacity style={styles.loginButton} onPress={doLogin}>
            <Text style={styles.textButton}>Login</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.registerButton} onPress={toRegister}>
            <Text style={styles.textButton2}>Create Account</Text>
          </TouchableOpacity>
        </>
      )}
    </View>
  )
}

export default Login
