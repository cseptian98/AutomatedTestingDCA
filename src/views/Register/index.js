import React, {useState, useEffect} from 'react'
import {
  Text,
  View,
  Image,
  TextInput,
  TouchableOpacity,
  StatusBar,
  BackHandler,
  Alert,
  ActivityIndicator,
} from 'react-native'
import styles from './Register.styles'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
import axiosConfig from 'api/BaseConfig'
import {StackActions} from '@react-navigation/routers'

const Register = ({navigation}) => {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [confirmationPassword, setConfirmPassword] = useState('')
  const [errorMessage, setErrorMessage] = useState('')
  const [isLoading, setIsLoading] = useState(false)
  const [showPassword, setShowPassword] = useState(true)
  const [showConfPassword, setShowConfPassword] = useState(true)

  const backAction = () => {
    const movePage = StackActions.pop(1)
    navigation.dispatch(movePage)
    return true
  }

  useEffect(() => {
    BackHandler.addEventListener('hardwareBackPress', backAction)
    return () => {
      BackHandler.removeEventListener('hardwareBackPress', backAction)
    }
  }, [])

  const doRegister = () => {
    if(password !== confirmationPassword) {
      setErrorMessage('Confirmation Password must same with Password')
    }
    const value = {email, password, confirmationPassword}
    setIsLoading(true)

    const onSuccess = () => {
      const registerSuccess = () => {
        setIsLoading(false)
        navigation.replace('Login')
      }
      Alert.alert('Register Success', 'Your Account Succesfully Registered', [
        {
          text: 'Oke',
          onPress: () => registerSuccess(),
        },
      ])
    }

    const onFailure = error => {
      console.log('debug error', error)
      Alert.alert('Register Error', errorMessage, [
        {
          text: 'Close',
          onPress: () => setIsLoading(false),
        },
      ])
    }

    axiosConfig
      .post('api/Auth/register', value)
      .then(onSuccess)
      .catch(onFailure)
  }

  const setIconPassword = () => {
    setShowPassword(!showPassword)
  }

  const setIconConfPassword = () => {
    setShowConfPassword(!showConfPassword)
  }

  return (
    <View style={styles.container}>
      <Image
        style={styles.image}
        source={require('assets/images/register.png')}
        testID="imageRegister"
      />

      <StatusBar style="auto" />
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Email"
          placeholderTextColor="#000"
          keyboardType="email-address"
          value={email}
          testID='emailRegister'
          onChangeText={(value) => setEmail(value)}
        />
      </View>

      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Password"
          placeholderTextColor="#000"
          secureTextEntry={showPassword}
          value={password}
          testID='passwordRegister'
          onChangeText={(value) => setPassword(value)}
        />
        <TouchableOpacity onPress={setIconPassword} style={styles.icon}>
          {showPassword ? (
            <Icon name='eye-off' size={24} color="#000" />
          ) : (
            <Icon name='eye' size={24} color="#000" />
          )}
        </TouchableOpacity>
      </View>

      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Confirmation Password"
          placeholderTextColor="#000"
          secureTextEntry={showConfPassword}
          value={confirmationPassword}
          testID='confirmPassword'
          onChangeText={(value) => setConfirmPassword(value)}
        />
        <TouchableOpacity onPress={setIconConfPassword} style={styles.icon}>
          {showConfPassword ? (
            <Icon name='eye-off' size={24} color="#000" />
          ) : (
            <Icon name='eye' size={24} color="#000" />
          )}
        </TouchableOpacity>
      </View>

      {isLoading ? (
        <ActivityIndicator size="large" color="#0D47A1" />
      ) : (
        <>
          <TouchableOpacity style={styles.registerButton} onPress={doRegister} testID='btnRegister'>
            <Text style={styles.registerText}>Register</Text>
          </TouchableOpacity>
        </>
      )}
    </View>
  )
}

export default Register
