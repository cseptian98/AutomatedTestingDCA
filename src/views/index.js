import HomeScreen from 'views/Home/Home'
import LoginScreen from 'views/Login/Login'
import RegisterScreen from 'views/Register/Register'

export {HomeScreen, LoginScreen, RegisterScreen}