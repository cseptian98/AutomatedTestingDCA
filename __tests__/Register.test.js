import React from 'react'
import {fireEvent, render, waitFor} from '@testing-library/react-native'
import {RegisterScreen} from 'views'
import {
  TEST_ID_IMAGE_REGISTER,
  TEST_ID_EMAIL_REGISTER,
  TEST_ID_PASSWORD_REGISTER,
  TEST_ID_CONFIRMATION_PASSWORD,
} from 'constants'

let componentRegister
let email
let password
let confirmPassword
let emailInput
let passwordInput
let confirmPasswordInput
const replace = jest.fn()
const params = {url: 'api/Auth/register'}

describe('Register Screen', () => {
  beforeEach(() => {
    componentRegister = render(
      <RegisterScreen route={{params}} navigation={{replace}} />,
    )

    email = 'cseptian@gmail.com'
    password = 'Auto@123'
    confirmPassword = 'Auto@123'

    emailInput = componentRegister.getByTestId(TEST_ID_EMAIL_REGISTER)
    passwordInput = componentRegister.getByTestId(TEST_ID_PASSWORD_REGISTER)
    confirmPasswordInput = componentRegister.getByTestId(
      TEST_ID_CONFIRMATION_PASSWORD,
    )
  })
  it('renders correctly', async () => {
    await waitFor(() => {
      const screen = componentRegister.toJSON()
      expect(screen).toMatchSnapshot()
    })
  })
  it('should show register screen', () => {
    expect(componentRegister.getByTestId(TEST_ID_IMAGE_REGISTER)).toBeTruthy()
  })

  it('should change form register', async () => {
    await waitFor(() => {
      fireEvent.changeText(emailInput, email)
      fireEvent.changeText(passwordInput, password)
      fireEvent.changeText(confirmPasswordInput, confirmPassword)
    })
    expect(emailInput.props.value).toBe(email)
    expect(passwordInput.props.value).toBe(password)
    expect(confirmPasswordInput.props.value).toBe(confirmPassword)
  })
})
