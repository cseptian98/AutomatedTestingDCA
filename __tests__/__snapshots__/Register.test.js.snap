// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/register.png",
      }
    }
    style={
      Object {
        "height": 200,
        "marginBottom": 16,
        "marginTop": 24,
        "width": 280,
      }
    }
    testID="TEST_ID_IMAGE_REGISTER"
  />
  <View
    style={
      Object {
        "borderBottomColor": "#212121",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#000"
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="TEST_ID_EMAIL_REGISTER"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#212121",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#000"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="TEST_ID_PASSWORD_REGISTER"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰈉
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#212121",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Confirmation Password"
      placeholderTextColor="#000"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="TEST_ID_CONFIRMATION_PASSWORD"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰈉
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0D47A1",
        "borderRadius": 16,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": "90%",
      }
    }
    testID="TEST_ID_BUTTON_REGISTER"
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontSize": 16,
        }
      }
    >
      Register
    </Text>
  </View>
</View>
`;
